src/containers/Home/index.tsx

```diff
@@ -1,7 +1,65 @@
 import React from 'react';
+import useForm, { isRequired, isEmail, maxChars } from '../../hooks/useForm';
 
 const Home: React.FC = () => {
-  return <h2>Home</h2>;
-}
+  const { fields, onChange, errors, disableForm } = useForm([
+    { field: 'name', value: '', validators: [isRequired] },
+    { field: 'email', value: '', validators: [isRequired, isEmail] },
+    { field: 'comment', value: '', validators: [maxChars(20)] },
+  ]);
+
+  const onSubmitHandler = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+  };
+
+  return (
+    <React.Fragment>
+      <h2>Home</h2>
+      <form onSubmit={onSubmitHandler}>
+        {/* Name field*/}
+        <label>Name: </label>
+        <br />
+        <input
+          type="text"
+          name="name"
+          onChange={onChange}
+          value={fields.name}
+        />
+        <ul>
+          {errors.name.map((error) => (
+            <li key={error}>Name {error}</li>
+          ))}
+        </ul>
+        {/* Email field*/}
+        <label>Email: </label>
+        <br />
+        <input
+          type="email"
+          name="email"
+          onChange={onChange}
+          value={fields.email}
+        />
+        <ul>
+          {errors.email.map((error) => (
+            <li key={error}>Email {error}</li>
+          ))}
+        </ul>
+        {/* Comment field*/}
+        <label>Comment: </label>
+        <br />
+        <textarea name="comment" onChange={onChange} value={fields.comment} />
+        <ul>
+          {errors.comment.map((error) => (
```

Comment:
Yes this will never be an empty array
I did the validation in the code 

>   const errorFunctionsByKey: Dictionary<[]> = useMemo(() => {
>     return createObjectFrom(
>       settings,
>       (item) => item.field,
>       (item) => **item.validators || []**
>     );
>   }, []);